Mathlib/CategoryTheory/Category/Preorder.lean

```diff
@@ -218,3 +218,14 @@ theorem Equivalence.toOrderIso_symm_apply (e : X ≌ Y) (y : Y) :
 end PartialOrder
 
 end CategoryTheory
+
+open CategoryTheory
+
+lemma PartialOrder.isIso_iff_eq {X : Type u} [PartialOrder X]
+    {a b : X} (f : a ⟶ b) : IsIso f ↔ a = b := by
+  constructor
+  · intro _
+    exact (asIso f).to_eq
+  · rintro rfl
+    obtain rfl : f = 𝟙 _ := rfl
```

Comment:
I think this should also be added as a simp lemma (for example above `homOfLe` in the same file):
```
@[simp]
theorem hom_self_eq_id {x : X} (f : x ⟶ x) : f = 𝟙 x := rfl
```

Reply:
Thanks for the suggestion!

Reply:
I have removed the `simp` attribute because it breaks things downstream.